Watchman.DomainModel/Settings/GetConfigurationQueryHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Watchman.Cqrs;
+using Watchman.Integrations.MongoDB;
+
+namespace Watchman.DomainModel.Settings
+{
+    public class GetConfigurationQueryHandler : IQueryHandler<GetConfigurationQuery, GetConfigurationQueryResult>
+    {
+        private readonly ISessionFactory _sessionFactory;
+
+        public GetConfigurationQueryHandler(ISessionFactory sessionFactory)
+        {
+            this._sessionFactory = sessionFactory;
+        }
+
+        public GetConfigurationQueryResult Handle(GetConfigurationQuery query)
+        {
+            using var session = this._sessionFactory.Create();
+            var allConfigurationItems = session.Get<ConfigurationItem>();
+            var serverConfigurationItems = allConfigurationItems.Where(x => x.ServerId == query.ServerId);
+            var defaultConfigurationItems = allConfigurationItems.Where(x => x.ServerId == 0);
+            if (!serverConfigurationItems.Any())
+            {
+                return new GetConfigurationQueryResult(defaultConfigurationItems);
+            }
+            foreach (var configurationItem in defaultConfigurationItems)
+            {
+                if (serverConfigurationItems.Any(x => x.Name == configurationItem.Name))
+                {
+                    continue;
+                }
+                serverConfigurationItems.Append(configurationItem);
+            }
```

Comment:
hmm może 
```csharp
serverConfigurationItems.AddRange(defaultConfigurationItems.Except(serverConfigurationItems))
```
zadziała? Na pewno jest czytelniejsze

Reply:
Wydaje mi się, że tak po prostu to nie zadziała, bo to są różne obiekty, a klas tak łatwo nie da się porównywać:
`If you want to compare sequences of objects of some custom data type, you have to implement the IEquatable<T> generic interface in a helper class.`
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.except?view=netcore-3.1